### PR TITLE
feat: Support the new iOS 14 picker style

### PR DESCRIPTION
**Pull request by mmazzarolo**
BREAKING CHANGE: Add support for the new iOS 14 picker style by using setting the "display" prop to "inline".
You could already set "display" to "inline" to achieve an iOS 14 style, but we tweaked a bit the modal padding to look better in this use case. This is a breaking change because if you were already using the "inline" display you might have to update your style customizations to account for the new padding.
We also added a new "pickerStyle" prop that can be used to customize the picker wrapper (which is the element we use to set the padding described above).

Also, we're starting to drop support for the iOS header:

BREAKING CHANGE: Do not show the built-in iOS header by default on iOS.
We are planning to stop the built-in iOS header support in the near future, but we'll still support custom header implementations passed to the "customHeaderIOS" prop.
For the time being, you can still show the built-in iOS header by setting the "isHeaderVisibleIOS" prop to true.

**With `display="spinner"` (the default)**

Simulator Screen Shot - iPhone 11 - 2021-05-22 at 12 47 14
Simulator Screen Shot - iPhone 11 - 2021-05-22 at 12 47 07
Simulator Screen Shot - iPhone 11 - 2021-05-22 at 12 47 17

**With `display="inline"`**

Simulator Screen Shot - iPhone 11 - 2021-05-22 at 12 46 58
Simulator Screen Shot - iPhone 11 - 2021-05-22 at 12 47 02
datetimepicker-ios-inline

### CODE REVIEW

File: example/ios/Podfile.lock [hunk 1]
```diff
@@ -296,7 +296,7 @@ PODS:
     - React-Core (= 0.63.2)
     - React-cxxreact (= 0.63.2)
     - React-jsi (= 0.63.2)
-  - RNDateTimePicker (2.6.1):
+  - RNDateTimePicker (3.0.2):
```
Comment (scott-massey) [inline example/ios/Podfile.lock:299]: don't you want to have it set to the latest version (3.4.7)?

File: src/DateTimePickerModal.ios.js [hunk 2]
```diff
@@ -189,6 +204,9 @@ const pickerStyles = StyleSheet.create({
   containerDark: {
     backgroundColor: BACKGROUND_COLOR_DARK,
   },
+  headerFiller: {
+    height: 8,
```
Comment (scott-massey) [inline src/DateTimePickerModal.ios.js:208]: I did `paddingTop: 14`, but this probably works too

**Review by scott-massey — COMMENTED** [review]
Love it! I had a couple comments but this definitely works for me. Do you think it would be better to default the new prop to `true` so it doesn't break other dev's apps? Obviously that's up to you though.

**Review by scott-massey — APPROVED** [review]

File: example/ios/Podfile.lock (hunk 1 above)
Comment (mmazzarolo) [reply example/ios/Podfile.lock:299]: This is just for the example app (which from the lock is depending on 3.0.2) 👍

**Comment by mmazzarolo** [comment]
Thanks for approving @scott-massey . Still thinking a bit about this because of the breaking changes.

**Comment by scott-massey** [comment]
You could probably just leave it defaulted to `true` for now, and then maybe in some future major version release switch it to default to `false`.

**Comment by mmazzarolo** [comment]
Yeah. My thought here is that if we're planning to release something like #570, we can directly avoid merging this to introduce fewer changes (with a single major bundle).

**Pull request merged**